queue-manager/rango-preset/src/helpers.ts

```diff
@@ -1109,71 +1108,74 @@ export function signTransaction(
     onFinish();
     return;
   }
-  const signer = walletSigners.getSigner(txType);
-  signer.signAndSendTx(tx, walletAddress, chainId).then(
-    ({ hash, response }) => {
-      const explorerUrl = getScannerUrl(
-        hash,
-        currentStepBlockchain,
-        meta.blockchains
-      );
-      setStepTransactionIds(
-        actions,
-        hash,
-        explorerUrl &&
-          (!response || (response && !response.hashRequiringUpdate))
-          ? { url: explorerUrl, description: isApproval ? 'Approve' : 'Swap' }
-          : undefined
-      );
-      // response used for evm transactions to get receipt and track replaced
-      response && setTransactionDataByHash(hash, { response });
-      schedule(SwapActionTypes.CHECK_TRANSACTION_STATUS);
-      next();
-      onFinish();
-    },
-    (error) => {
-      if (swap.status === 'failed') {
-        return;
-      }
 
-      const { extraMessage, extraMessageDetail, extraMessageErrorCode } =
-        prettifyErrorMessage(error);
+  getSigners(sourceWallet.walletType).then((walletSigners) => {
```

Comment:
I think making `signTransaction` async and use async/await here would be better:
1. It communicates it runs something asynchronously explicitly
2. History for these lines will be kept.

Reply:
Done in [fix: make signTransaction async in rango-preset](https://github.com/rango-exchange/rango-client/pull/861/commits/b36e3181cbb33803fd2d65120387ae079b0df5c8)